core/state/state_object.go

```diff
@@ -268,16 +254,13 @@ func (s *stateObject) SetState(db Database, key, value common.Hash) {
 // lookup only happens in the fake state storage.
 //
 // Note this function should only be used for debugging purpose.
-func (s *stateObject) SetStorage(storage map[common.Hash]common.Hash) {
-	// Allocate fake storage if it's nil.
-	if s.fakeStorage == nil {
-		s.fakeStorage = make(Storage)
-	}
-	for key, value := range storage {
-		s.fakeStorage[key] = value
+//
+//@deprecated use SetState per item instead. This method no longer fully replaces
+// all slots.
```

Comment:
I'm not following this deprecation warning? It didn't fully replace all the slots before either.

Reply:
Can't we just delete this altogether if there's a better alternative?